### ISSUE TITLE
agent: Runtime error on a customer device

### ISSUE THREAD

**Issue by otavio** [shellhub-io/shellhub#183]
```
[root@xxx ~]# shellhub-agent
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x909c04]

goroutine 1 [running]:
github.com/shellhub-io/shellhub/pkg/api/client.buildURL(0x1c0d700, 0x90fc4f, 0x5, 0x0, 0x1d4c840)
        /home/xxx/solar/new/xxx-yocto-platform/build/tmp/work/cortexa7hf-neon-vfpv4-jel-linux-gnueabi/shellhub-agent/v0.3.2-r0/build/src/github.com/shellhub-io/shellhub/pkg/api/client/client.go:100 +0xfc
github.com/shellhub-io/shellhub/pkg/api/client.(*client).GetInfo(0x1c0d700, 0x1, 0x1, 0xa69b90)
        /home/xxx/solar/new/xxx-yocto-platform/build/tmp/work/cortexa7hf-neon-vfpv4-jel-linux-gnueabi/shellhub-agent/v0.3.2-r0/build/src/github.com/shellhub-io/shellhub/pkg/api/client/client_public.go:27 +0x48
main.main()
        /home/xxx/solar/new/xxx-yocto-platform/build/tmp/work/cortexa7hf-neon-vfpv4-jel-linux-gnueabi/shellhub-agent/v0.3.2-r0/build/src/github.com/shellhub-io/shellhub/agent/main.go:74 +0x308
```

**Comment by otavio** [comment]
This happens when variables are not set. Please adjust it to provide a proper error message which helps user to know what is going on.

**Comment by shawn111** [comment]
@otavio  https://github.com/shellhub-io/shellhub/pull/617

How about let envconfig to handle required env?

I did a few change in main().

```
 type ConfigOptions struct { runtime error: invalid memory address or nil pointer dereference
	ServerAddress string `envconfig:"server_address" required:"true"`  // envconfig would show error instead of "runtime error: invalid memory address or nil pointer dereference"
	PrivateKey string `envconfig:"private_key" required:"true"`
	TenantID string `envconfig:"tenant_id" required:"true"`
...

	opts := ConfigOptions{}
+	defer envconfig.Usage("shellhub", &opts)                                // present how the config works for user
	// Process unprefixed env vars for backward compatibility
+	envconfig.Process("", &opts)
-	if err := envconfig.Process("", &opts); err != nil {
-		logrus.Panic(err)
-	}

	if err := envconfig.Process("shellhub", &opts); err != nil {
		logrus.Panic(err)
	}
```

envconfig.Process("", &opts) & envconfig.Process("shellhub", &opts) are useful.
but we don't need to handle first error, otherwise no chance to prcoess with "shellhub".

**Comment by otavio** [comment]
Can you propose a PR to address this?

**Comment by shawn111** [comment]
https://github.com/shellhub-io/shellhub/pull/617/files

Yes, I did. But not sure how to link issue and PR.

**Comment by otavio** [comment]
Oh, it is mixed on the commit. Could you split it out on a specific commit and open a PR?

**Comment by otavio** [comment]
Fixed in #621.
